src/App.tsx

```diff
@@ -17,18 +50,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onStatusChange={setStatus}
+                onQueryChange={setQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todosFromServer.length === 0 && <Loader />}
```

Comment:
looks cleaner this way to me. No strong opinion though
```suggestion
              {!todosFromServer.length && <Loader />}
```